### PR TITLE
Added scala dev guide

### PR DESCRIPTION
**Pull request by ahlag**
Signed-off-by: Chang Yong Lik <theeahlag@gmail.com>

## Description
- Added Scala Dev Guide for newcomers

## How was this patch tested?
None

## Does this PR introduce any user-facing changes?
User guide in docs/dev_guide (Similar to python_dev_guide.md)

### CODE REVIEW

**Review by xiaoyongzhu — APPROVED** [review]

**Comment by ahlag** [comment]
@xiaoyongzhu 
Can I ask you the following questions?
1. Is there a scala style guide?
1. Do you know how to execute a single test within and a single test suite? I have tried the following command but failed.
`sbt 'testOnly com.linkedin.feathr.offline.AnchoredFeaturesIntegTest.tesSWAWithPreprocessing'`
I have listed all the questions I have for you as TBD in the PR.

**Comment by xiaoyongzhu** [comment]
> @xiaoyongzhu Can I ask you the following questions?
> 
> 1. Is there a scala style guide?
> 2. Do you know how to execute a single test within and a single test suite? I have tried the following command but failed.
>    `sbt 'testOnly com.linkedin.feathr.offline.AnchoredFeaturesIntegTest.tesSWAWithPreprocessing'`
>    I have listed all the questions I have for you as TBD in the PR.

I'm not super familiar with that part TBH. looping in @jaymo001 and @hangfei to chime in

File: docs/dev_guide/scala_dev_guide.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+---
+
+# Feathr Scala Project Developer Guide
+
+## IntelliJ Setup
+
+IntelliJ is the recommended IDE to use when developing Feathr. Please visit IntelliJ's 
+[installation guide](https://www.jetbrains.com/help/idea/installation-guide.html) to install it
+in your local machine. To import Feathr as a new project:
+1. Git clone Feathr into your local machine.
```
Comment (hangfei) [inline docs/dev_guide/scala_dev_guide.md:14]: add git clone cmd here.

File: docs/dev_guide/scala_dev_guide.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+3. Under `Import project from external model` select `sbt`. Click `Next`.
+4. Under `Project JDK` specify a valid Java `1.8` JDK and select SBT shell for `project reload` and `builds`.
+5. Click `Finish`.
+6. You should see something like `[success] Total time: 5 s, completed Jun 1, 2022 9:43:26 PM` in sbt shell.
+
+### Setup Verification
+
+After waiting for IntelliJ to index, verify your setup by running a test suite in IntelliJ.
+
+1. Search for and open `AnchoredFeaturesIntegTest`
```
Comment (hangfei) [inline docs/dev_guide/scala_dev_guide.md:25]: this may change. so let's show devs how to find a random test.

**Review by hangfei — COMMENTED** [review]
Overall looks pretty good. Developer experience is one of the most critical things we should keep improving.

Thanks for your work!

**Comment by ahlag** [comment]
@hangfei 
Ok done!

**Review by hangfei — APPROVED** [review]

**Pull request merged**